java/test/org/openqa/selenium/remote/http/RetryRequestTest.java

```diff
@@ -96,12 +96,11 @@ void canReturnAppropriateFallbackResponse() {
   @Test
   void canReturnAppropriateFallbackResponseWithMultipleThreads()
       throws InterruptedException, ExecutionException {
-    HttpHandler handler1 =
-        new RetryRequest()
-            .andFinally(
-                (HttpRequest request) -> {
-                  throw new TimeoutException();
-                });
+    AppServer server = new NettyAppServer(req -> new HttpResponse());
+
+    URI uri = URI.create(server.whereIs("/"));
+    HttpRequest connectionTimeoutRequest =
+        new HttpRequest(GET, String.format(REQUEST_PATH, uri.getHost(), uri.getPort()));
```

Comment:
This was hard for me to understand, the server is not started and therefore a `ConnectException` is raised.
So the server is only to get a url where a connection will fail.

Would it be possible to replace the `throw new TimeoutException();` with a `throw new ConnectException()` and keep the handler1?

Reply:
Yes, I did not want to use a random URL which might fail. So used the mechanism we have already to simulate the connection failure. 
Using `throw new ConnectException() ` was my first instinct but that is a checked exception and requires adding handling it in the HttpHandler class. That did not seem right to add because a test was throwing the error.

Reply:
I'm taking the thumbs-up to mean this is good to go, so I'm merging it. :)